Separated bug and feature issue templates

galexrt opened the pull request:
Description of your changes:
This enables us to have a "template" selection of either bug report or feature request when a user creates an issue. This improves the issue creation flow by removing the need to remove certain sections depending on which type of issue the user has.

An example repository with slightly modified issue templates as a demo can be found here: https://github.com/galexrt/github-issue-templates. Just go to `issues`->`New issue` and you should get this issue type/template prompt:
![2018-06-02-151616_1273x561_scrot](https://user-images.githubusercontent.com/3718398/40874973-fde289b4-6677-11e8-968a-61fd5feb3b8d.png)

Note: It is also possible to create more types than just bug report and feature request. Additionally a user can also "skip" the selection and get the current issue template.

Which issue is resolved by this Pull Request:
Resolves #

Checklist:
- [x] Documentation has been updated, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.

galexrt commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,30 @@
+
+Does this have an impact on existing features:
+
+**Environment**:
```
Like that?

tonya11en commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,29 @@
+
+**Bug Report**
+
+What happened:
```
I think these fields should be bold or italics to easily distinguish it from the answers at a glance.

tonya11en replied:
I would also rephrase this to be "Expected behavior" and "Deviation from expected behavior" or something similar. It just sounds more professional.

tonya11en commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,29 @@
+* Rook version (use `rook version` inside of a Rook Pod):
+* Kubernetes version (use `kubectl version`):
+* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
+* Ceph status (use `ceph health` in the [Rook toolbox](https://Rook.io/docs/Rook/master/toolbox.html)):
```
Change this to not be Ceph-specific. There will be other backends.

tonya11en commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,24 @@
+
+What would be solved through this feature:
+
+Does this have an impact on existing features:
```
How would we expect non-developers to know this?

galexrt replied:
This is just meant when a user knows if there is an impact that he can directly add it.

tonya11en replied:
But why would a user know such a thing and not the developers who would implement it? This is something that can't be known until there's a design in place for the implementation. We can't expect anyone to know this at the time of feature request. I think this is an unnecessary field.

tonya11en commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,24 @@
+
+**Feature Request**
+
+Are there any similar features already existing:
```
Is this really necessary? Why not just have a disclaimer at the top saying something like "Verify there are not similar features already in existence before filing the feature request."

galexrt replied:
I'd like to leave it in because if there are similar features we can see if we can a) build upon them or b) if it is not necessary to "implement" the request because of the existing feature.

tonya11en replied:
I'm not convinced- it still seems unnecessary.

In the case of (a), that decision should be made by developers that would implement the feature. The developers should know what features exist already and shouldn't be relying on someone filing a ticket to remind them of things they could extend to get the desired effect. In the case of (b), it would be asinine to even file a feature request for something that already exists.

tonya11en commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,24 @@
+
+Are there any similar features already existing:
+
+What should the feature do:
```
You can probably collapse this and the next one into "Describe the feature and the problem it will solve." I see no need for two separate fields.

galexrt replied:
Depends on how you see it. I see those fields as a separate:
1. `What should the feature do`: What the feature really does, e.g. run Ceph as a storage backend.
2. `What would be solved through this feature`: What the feature "actually" solves, e.g. world storage needs are solved by adding Ceph as storage backend.

tonya11en replied:
So you mean for (2) to be a "Motivation" or "Use case" field. It should be renamed as such.

galexrt commented:
@tonya11en Added your feedback. PTAL

tonya11en commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -3,29 +3,29 @@
+**Deviation from expected behavior:**
 
-What you expected to happen:
+**Expected behavior:**
```
This is a nit, but I'd switch the location of this so that it comes before the deviation portion.

tonya11en commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -2,12 +2,12 @@
 
-Checklist:
+**Checklist:**
 - [ ] Documentation has been updated, if necessary.
```
Also a nit, but I'd make these checklist items less wordy. Something like:

**Checklist:**
If necessary:
- [ ] Documentation updated.
- [ ] Pending release notes updated.
- [ ] Upgrade tested.


I don't see a need for all of the extras. For instance, the user guide is documentation so you don't need to mention that again. It should also be obvious to any competent developer that the pending release notes are updated with any notable changes or changes that would break other features.

galexrt commented:
@jbw976 @travisn @tonya11en I have updated the templates (also added another point about `make vendor` to the PR checklist).
Please let me know if that is better now.

jbw976 commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -2,12 +2,14 @@
 - [ ] Documentation has been updated, if necessary.
 - [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
 - [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
+- [ ] Code generation has been run to update object specifications.
```
include the command here? `make codegen`

galexrt replied:
@jbw976 Done.

galexrt commented:
@tonya11en The only outstanding point I see is https://github.com/rook/rook/pull/1764/files#r193821723, but I'm on @travisn side their. The checklist reminds me most of the time to double check, e.g. if I updated the docs for a new feature.
I personally don't see an issue with having four of the points end with `if necessary`, in light of @jbw976 approval it seems that it is okay for him.
